crates/core/src/contracts/erc20.rs

```diff
@@ -0,0 +1,50 @@
+use std::marker::PhantomData;
+
+use reth_primitives::U256;
+use starknet::core::types::{BlockId, FunctionCall};
+use starknet::providers::Provider;
+use starknet_crypto::FieldElement;
+
+use crate::client::constants::selectors::BALANCE_OF;
+use crate::client::errors::EthApiError;
+use crate::client::helpers::DataDecodingError;
+use crate::models::felt::Felt252Wrapper;
+
+pub struct Erc20Contract<P> {
```

Comment:
can you add rustdoc to explain this is an abstraction over erc20.sol thru kakarot? to dispell misunderstandings

Reply:
Good that you mention that because looking more into it, I wasn't implementing Erc20 but Eth.